CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc

```diff
@@ -57,6 +57,7 @@ static function CHEventListenerTemplate CreateStrategyListeners()
 	Template.AddCHEvent('ShouldCleanupCovertAction', ShouldCleanupCovertAction, ELD_Immediate);
 	Template.AddCHEvent('OnResearchReport', TriggerPrototypeAlert, ELD_OnStateSubmitted);
 	Template.AddCHEvent('OnResearchReport', CheckTechRushCovertActions, ELD_OnStateSubmitted);
+	Template.AddCHEvent('OnResearchCompletePopupClosed', CheckTechRushCovertActions, ELD_OnStateSubmitted);
```

Comment:
There is `TriggerEvent('ResearchCompleted', TechState, ResearchProject, AddToGameState)`, which should handle any cases cleanly - including different types of techs (research/pg/shadow chamber) repeatable or not, etc